app/services/jobseekers/job_applications/employment_gap_finder.rb

```diff
@@ -0,0 +1,64 @@
+class Jobseekers::JobApplications::EmploymentGapFinder
+  def initialize(job_application)
+    @job_application = job_application
+    @today = Time.zone.today
+  end
+
+  def significant_gaps(threshold: 3.months)
+    job_application.employments.each_with_object({}) do |employment, gaps|
+      next if employment.current_role?
+      next if gap_to_today_is_less_than_threshold?(employment, threshold)
+      next if overlapping_employment?(employment)
+      next if next_employment_started_within_threshold?(employment, threshold)
+
+      gaps[employment] = {
+        started_on: employment.ended_on + 1.day,
+        ended_on: gap_end_date(employment) || today,
+      }
+    end
+  end
+
+  private
+
+  attr_reader :job_application, :today
+
+  def gap_to_today_is_less_than_threshold?(employment, threshold)
+    employment.ended_on + threshold >= today
+  end
+
+  def overlapping_employment?(employment)
+    job_application.employments.any? do |other|
```

Comment:
Could you do something like `.excluding(employment)` here?

Could save on the extra line `next if employment == other`

Reply:
@josephhull676 This is not a bad idea, but having tried it just now the problem is that only works if all the records are saved to the database, whereas the current implementation will work on any array of employments.

Reply:
Ahh ok 👍